fccodegen/_generator.py

```diff
@@ -9,16 +9,45 @@
 import time
 
 
+# Shared Jinja environment
+_environment = None
+
+
+def _jinja_environment():
+    """
+    Returns a shared Jinja environment to create templates from.
+    """
+    global _environment
+    if _environment is None:
+        _environment = jinja2.Environment(
+            # Automatic loading of templates stored in the module
+            # This also enables template inheritance
+            loader=jinja2.PackageLoader('fccodegen', 'templates'),
+
+            # Keep a single trailing newline, if present
+            keep_trailing_newline=True,
+
+            # Don't replace undefined template variables by an empty string
+            # but raise a jinja2.UndefinedError instead.
+            undefined=jinja2.StrictUndefined,
+        )
+    return _environment
+
+
 def load_template(*name):
     """
     Returns a path to the given template
     """
-    path = os.path.join(cg.DIR_TEMPLATE, *name)
-    #TODO: Check absolute path is still in DIR_TEMPLATE
+    path = os.path.join(*name)
+    #TODO: Check absolute path is still in cg.DIR_TEMPLATE
+
+    env = _jinja_environment()
+    return env.get_template(path)
+
     #TODO: Check exists
     #TODO: Look at the jinja PackageLoader. What's the advantage?
-    with open(path, 'r') as f:
-        return jinja2.Template(f.read())
+    #with open(path, 'r') as f:
```

Comment:
Get rid of these rather than commenting (and the TODO above :) )

Reply:
ok!